### ISSUE TITLE
Tapping “source code wrap” button permits quotes in code to be changed to curly quotes

### ISSUE THREAD

**Issue by benguild** [GitHawkApp/GitHawk#2316]
I pasted some code into a comment and tapped the code wrap button for backticks, but the quotes all flipped.

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.22.0 (1534395229)
Device: iPhone X (iOS 12.1)
TestFlight: false
</pre>
</details>

**Comment by rnystrom** [comment]
Can you give us some steps on how exactly to reproduce this?

**Comment by benguild** [comment]
Just paste in something like this:
```javascript
<script>var test = ""</script>
```

Highlight it, then press the `</>` button. If it doesn't do it, highlight it again and do it again, or enter some text around it and try it again.

**Comment by BasThomas** [comment]
Can you check if you have Smart Punctuation turned on in Settings > General > Keyboard?

**Comment by rnystrom** [comment]
Doesn't repro for me

![simulator screen shot - iphone xs - 2018-10-20 at 17 49 38](https://user-images.githubusercontent.com/739696/47260835-aac1b200-d490-11e8-8e9f-eb2de6b86393.png)

![simulator screen shot - iphone xs - 2018-10-20 at 17 50 03](https://user-images.githubusercontent.com/739696/47260838-b1502980-d490-11e8-9b8f-3e3f06c746f1.png)

![simulator screen shot - iphone xs - 2018-10-20 at 17 49 41](https://user-images.githubusercontent.com/739696/47260837-aeedcf80-d490-11e8-8edd-879325200b01.png)

**Comment by benguild** [comment]
@rnystrom I think the quotes are already angled in your screenshots.

**Comment by benguild** [comment]
@BasThomas Yes, "Smart Punctuation" is enabled as it's generally the default, no? I had an issue with this on a macOS editor once but they were able to resolve it without affecting other word processors/text input except when dealing with code, and this is probably the way to go.

**Comment by rnystrom** [comment]
I haven’t changed my kb from the factory setting 🤷‍♀️

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
I mean we can get a reference to the text, then reiterate over it and replace previously normal quotes that became angled quotes with normal ones if they are within a code block?

That sounds like a lot of effort though. 😛

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by benguild** [comment]
I'm not sure but regardless it seems like smart punctuation is going to cause problems on macOS or iOS when dealing with files that are code or in markdown with embedded code.

**Comment by benguild** [comment]
This also happens when pasting in HTML from other issues:

![img_0012](https://user-images.githubusercontent.com/1185659/48995197-4a5a1a80-f18a-11e8-90e3-61490239d301.jpg)

**Comment by Huddie** [comment]
Pretty sure this issue was mentioned in the past

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by benguild** [comment]
One possible solution would be to have a markdown editor mode for posts
that does not do this. Unless you for sure want fancy punctuation, it’s
safer to turn off for anything that supports markdown or html.
-- 
iPhoneから送信
